commit 793e721d21fd7d2825e3b7e56ff5009ec6d052e9
Author: Christian Lamparter <chunkeey@googlemail.com>
Date:   Sun Sep 26 21:28:51 2010 +0200

    carl9170: remove non-standard ba session teardown
    
    802.11n-2009 demands in 11.2.1: "
    When a STA enters normal (non-APSD) PS mode, any downlink
    Block ACK agreement without an associated schedule is
    suspended for the duration of this PS mode."
    
    The operative word is "suspended" and not terminated.
    
    Signed-off-by: Christian Lamparter <chunkeey@googlemail.com>
    Signed-off-by: John W. Linville <linville@tuxdriver.com>

--- a/drivers/net/wireless/ath/carl9170/main.c
+++ b/drivers/net/wireless/ath/carl9170/main.c
@@ -1453,9 +1453,6 @@ static void carl9170_op_sta_notify(struc
 				while ((skb = __skb_dequeue(&tid_info->queue)))
 					__skb_queue_tail(&free, skb);
 				spin_unlock_bh(&tid_info->lock);
-
-				ieee80211_stop_tx_ba_session(sta,
-					tid_info->tid);
 			}
 			rcu_read_unlock();
 		}
